SourceryTests/ParserSpec.swift

```diff
@@ -496,6 +496,74 @@ class ParserSpec: QuickSpec {
                                                   ])
                                           ]))
                     }
+
+                    context("given associated value with its type existing") {
+
+                        it("extracts associated value's type") {
+                            let types = parse("protocol Baz {}; class Bar: Baz {}; enum Foo { case optionA(key: Bar) }")
+                            let enumeration = types[2] as? Enum
+                            let enumCases = enumeration?.cases
+                            expect(enumCases?.count) == 1
+
+                            let associatedValues = enumCases?.first?.associatedValues
+                            expect(associatedValues?.count) == 1
+
+                            let associatedValue = associatedValues?.first
+                            expect(associatedValue?.name) == "key"
+                            expect(associatedValue?.typeName) == "Bar"
```

Comment:
You can use equality against whole type like we do in other tests, this has benefits because we diff data and it will show us better errors if test fail